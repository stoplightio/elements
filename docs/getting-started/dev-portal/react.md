# Elements Dev Portal in React

Learn how to quickly get started with Elements Dev Portal in a React project.

## Create React App Template

We've created a Create React App template, which allows you to create a brand new Elements Dev Portal website in React without any additional setup.

Simply run:

`npx create-react-app my-dir --template @stoplight/elements-dev-portal`.

Then run `cd my-dir` and `yarn start` and you will see a basic Elements Dev Portal website in the browser.

## Usage

1. First, install the elements-dev-portal package with NPM/Yarn.

```bash
yarn add @stoplight/elements-dev-portal
```

2. In `App.js` import the API component and CSS file from the Elements library.

```jsx
import { StoplightProject } from '@stoplight/elements-dev-portal';
import '@stoplight/elements-dev-portal/styles.min.css';
```

3. Find the "Project ID" from the Project Settings view of your Stoplight Project.

![The product ID can be found on the Project Settings page in a text box after Display Name and Slug](../../images/projectId.png)


4. Now you can replace the existing App component's contents with the StoplightProject component.

```jsx
<StoplightProject projectId="cHJqOjYyNTgw" />
```

Now your `App.js` file should look something like this:

<!-- title: App.js -->
```jsx
import React from 'react';

import { StoplightProject } from '@stoplight/elements-dev-portal';
import '@stoplight/elements-dev-portal/styles.min.css';


function App() {
  return (
    <div className="App">
      <StoplightProject projectId="cHJqOjYyNTgw" />
    </div>
  );
}

export default App;
```

## Fire it up

Now start the development server.

```bash
yarn start
```

And you should see the API reference documentation for the Zoom API!

## Configuration

- `projectId` - Find the Project ID in the Project Settings screen on Stoplight.
- `platformUrl` - By default this is `https://stoplight.io/` and this default will be fine for most people. Some enterprise customers might need to change it.
- `basePath` - Helps when using `router: 'history'` but docs are in a subdirectory like `https://example.com/docs/api`.
- `hideMocking` - Pass `true` to disable all mocking options and remove mention of the mock server.
- `hideTryIt` - Pass `true` to hide the "Try It" panel (the interactive API console).
- `router` -  Determines how navigation should work:
  - `history` - (default) uses the HTML5 history API to keep the UI in sync with the URL.
  - `hash` - uses the hash portion of the URL (i.e. window.location.hash) to keep the UI in sync with the URL.
  - `memory` - keeps the history of your "URL" in memory (does not read or write to the address bar).
