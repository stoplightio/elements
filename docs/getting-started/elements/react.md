# Getting Started with Elements in React

Learn how to quickly get started with Elements in a React project.

## Create React App Template

We've created a [Create React App template](https://github.com/stoplightio/cra-template-elements), which allows you to create a brand new Elements website in React without any additional setup.

> Note: [The Create React App template only works with version 4 of create-react-app because of Webpack 5 polyfill issues.](https://github.com/facebook/create-react-app/issues/11756)

```bash
npx create-react-app@4.0.3 my-dir --template @stoplight/elements
```

Then run `cd my-dir` and `yarn start` and you will see a basic Elements website in the browser.

## Manual Setup

To install Elements Dev Portal in an existing React app, follow these instructions.


1. Install the [@stoplight/elements](https://www.npmjs.com/package/@stoplight/elements) package with NPM/Yarn.

```bash
yarn add @stoplight/elements
```

2. In `App.js` import the API component and CSS file from the Elements library.

```jsx
import { API } from '@stoplight/elements';
import '@stoplight/elements/styles.min.css';
```

1. Add the App component to the output of the app.

```jsx
<API
  apiDescriptionUrl="https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml"
/>
```

4. Now your `App.js` file should look something like this:

<!-- title: App.js -->
```jsx
import React from 'react';

import { API } from '@stoplight/elements';
import '@stoplight/elements/styles.min.css';


function App() {
  return (
    <div className="App">
      <API
        apiDescriptionUrl="https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml"
      />
    </div>
  );
}

export default App;
```
## Fire it up

Now start the development server.

```bash
yarn start
```

And you should see the API reference documentation for the Zoom API!

## Configuration

See [Elements Configuration Options](elements-options.md). 

## Examples

<!-- title: React Component with API Description Provided as a URL -->
```
<API
  apiDescriptionUrl="https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml"
  router="hash"
/>
```

<!-- title: React Component with API Description Provided Directly -->

```jsx
import { API } from "@stoplight/elements";

const apiDescriptionDocument = {
  openapi: '3.1.0',
  info: {
    title: 'Some Awesome API',
    version: '1.0.0'
  },
  paths: {
    /* ... */
  }
};

<API
  apiDescriptionDocument={apiDescriptionDocument}
  router="hash"
/>
```
