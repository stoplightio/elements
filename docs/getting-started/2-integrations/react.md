# Getting Started with Elements in React

Learn how to quickly get started with Elements in a new React project.

## Create the React App

Run the following command to create a starter React app using [create-react-app](https://github.com/facebook/create-react-app).

```bash
npx create-react-app elements-starter-react
cd elements-starter-react
```

## Install Elements

Next, install the Elements library and it's peer dependencies.

```bash
yarn add @stoplight/elements @stoplight/prism-http mobx
```

In `App.js` import the API component and CSS file from the Elements library.

<!-- title: App.js -->
```jsx
import { API } from '@stoplight/elements';
import '@stoplight/elements/styles.min.css';
```

Now you can replace the existing App component's contents with the API component we just imported.

<!-- title: App.js -->
```jsx
<API
  apiDescriptionUrl="https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml"
/>
```

> Check out the [API component docs](../components/API.md) to see what other properties are avaiable.

Now your `App.js` file should look something like this:

<!-- title: App.js -->
```jsx
import React from 'react';

import { API } from '@stoplight/elements';
import '@stoplight/elements/styles.min.css';


function App() {
  return (
    <div className="App">
      <API
        apiDescriptionUrl="https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml"
      />
    </div>
  );
}

export default App;
```

Simple, huh?

## Fire it up

Now start the development server.

```bash
yarn start
```

And you should see the API reference documentation for the Zoom API.

![](https://cdn.stoplight.io/elements/elements-starter-react-zoom-api-reference-docs.png)


## Next steps

Well that was easy, wasn't it? But you're not done yet! Elements provides you with the components for beautiful API reference documentation, but you'll likely want to add other things to your site such as a landing page, header and footer navigations, etc.

If you need some inspiration, check out some of our examples:

- [elements-starter-gatsby](https://github.com/stoplightio/elements-starter-gatsby)
- [elements-starter-react](https://github.com/stoplightio/elements-starter-react)

Once you're done and you're docs are live, give us a shout [@stoplightio](https://twitter.com/stoplightio) and we'll help you share it with the world!
