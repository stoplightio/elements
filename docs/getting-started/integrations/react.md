# Getting Started with Elements in React

Learn how to quickly get started with Elements in a React project.

## Using Elements in ReactJS

First, install the Elements package with NPM/Yarn.

```bash
yarn add @stoplight/elements
```

In `App.js` import the API component and CSS file from the Elements library.

```jsx
import { API } from '@stoplight/elements';
import '@stoplight/elements/styles.min.css';
```

Now you can replace the existing App component's contents with the API component we just imported.

```jsx
<API
  apiDescriptionUrl="https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml"
/>
```

Now your `App.js` file should look something like this:

<!-- title: App.js -->
```jsx
import React from 'react';

import { API } from '@stoplight/elements';
import '@stoplight/elements/styles.min.css';


function App() {
  return (
    <div className="App">
      <API
        apiDescriptionUrl="https://raw.githubusercontent.com/stoplightio/Public-APIs/master/reference/zoom/openapi.yaml"
      />
    </div>
  );
}

export default App;
```

## Fire it up

Now start the development server.

```bash
yarn start
```

And you should see the API reference documentation for the Zoom API!

## Configuration

- `apiDescriptionUrl` - OpenAPI document URL, supporting `http://`, `https://`, and documents containing `$ref` to other http(s) documents.
- `apiDescriptionDocument` - OpenAPI document, provided as YAML string, JSON string or JavaScript object.
- `basePath` - Helps when using `router: 'history'` but docs are in a subdirectory like `https://example.com/docs/api`.
- `hideTryIt` - Pass `true` to hide the "Try It" panel (the interactive API console).
- `layout` - There are two layouts for Elements:
  - `sidebar` - (default) Three-column design.
  - `stacked` - Everything in a single column, making integrations with existing websites that have their own sidebar or other columns already.
- `logo` - URL to an image that will show as a small square logo next to the title, above the table of contents.
- `router` -  Determines how navigation should work:
  - `history` - (default) uses the HTML5 history API to keep the UI in sync with the URL.
  - `hash` - uses the hash portion of the URL (i.e. window.location.hash) to keep the UI in sync with the URL.
  - `memory` - keeps the history of your "URL" in memory (does not read or write to the address bar).

## Examples

-<!-- title: React Component with API Description Provided as a URL -->

<API
  apiDescriptionUrl="https://raw.githubusercontent.com/stoplightio/Public-APIs/master/eference/zoom/openapi.yaml"
  router="hash"
/>


<!-- title: React Component with API Description Provided Directly -->

```jsx
import { API } from "@stoplight/elements";

const apiDescriptionDocument = {
  openapi: '3.1.0',
  info: {
    title: 'Some Awesome API',
    version: '1.0.0'
  },
  paths: {
    /* ... */
  }
};

<API
  apiDescriptionDocument={apiDescriptionDocument}
  router="hash"
/>
```
