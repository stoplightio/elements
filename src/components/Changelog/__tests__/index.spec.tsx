import 'jest-enzyme';

import { mount, ReactWrapper } from 'enzyme';
import * as React from 'react';

import { Changelog } from '../index';

describe('Changelog', () => {
  let wrapper: ReactWrapper;

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render null with no changes', () => {
    wrapper = mount(<Changelog changes={undefined} />);

    expect(wrapper).toHaveText('No changes for this resource.');
  });

  it('should render null with empty changes', () => {
    wrapper = mount(<Changelog changes={[]} />);

    expect(wrapper).toHaveText('No changes for this resource.');
  });

  it('should match snapshot', () => {
    wrapper = mount(
      <Changelog
        changes={[
          {
            createdAt: '1578032747722',
            message: 'major message',
            semver: 'major',
          },
          {
            createdAt: '1578032747722',
            message: 'minor message',
            semver: 'minor',
          },
          {
            createdAt: '1577946373272',
            message: 'patch message',
            semver: 'patch',
          },
        ]}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
