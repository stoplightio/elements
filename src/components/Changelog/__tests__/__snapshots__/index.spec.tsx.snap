// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changelog should match snapshot 1`] = `
<Memo()
  changes={
    Array [
      Object {
        "createdAt": "1578032747722",
        "message": "major message",
        "semver": "major",
      },
      Object {
        "createdAt": "1578032747722",
        "message": "minor message",
        "semver": "minor",
      },
      Object {
        "createdAt": "1577946373272",
        "message": "patch message",
        "semver": "patch",
      },
    ]
  }
>
  <div
    className="Changelog"
  >
    <div
      className=""
      key="Fri Jan 03 2020"
    >
      <div
        className="Changelog__date p-2 font-medium"
      >
        Fri Jan 03 2020
      </div>
      <div
        className="Changelog__item p-2 border-b border-gray-2 dark:border-lighten-3"
        key="0"
      >
        <div
          className="flex text-sm"
        >
          <div
            className="Changelog__message flex-1 mr-3"
          >
            major message
          </div>
          <div
            className="Changelog__semver lowercase font-semibold"
          >
            major
          </div>
        </div>
      </div>
      <div
        className="Changelog__item p-2 border-b border-gray-2 dark:border-lighten-3"
        key="1"
      >
        <div
          className="flex text-sm"
        >
          <div
            className="Changelog__message flex-1 mr-3"
          >
            minor message
          </div>
          <div
            className="Changelog__semver lowercase font-semibold"
          >
            minor
          </div>
        </div>
      </div>
    </div>
    <div
      className="mt-6"
      key="Thu Jan 02 2020"
    >
      <div
        className="Changelog__date p-2 font-medium"
      >
        Thu Jan 02 2020
      </div>
      <div
        className="Changelog__item p-2 border-b border-gray-2 dark:border-lighten-3"
        key="0"
      >
        <div
          className="flex text-sm"
        >
          <div
            className="Changelog__message flex-1 mr-3"
          >
            patch message
          </div>
          <div
            className="Changelog__semver lowercase font-semibold"
          >
            patch
          </div>
        </div>
      </div>
    </div>
  </div>
</Memo()>
`;
